optimum/bettertransformer/transformation.py

```diff
@@ -0,0 +1,190 @@
+    sort_fn = lambda list_modules: [module.__class__.__name__ for module in list_modules]  # noqa: E731
+
+    for key in dict_named_module.keys():
+        if isinstance(dict_named_module[key], torch.nn.ModuleList) and all(
+            "LayerBetterTransformer" in module_name for module_name in sort_fn(dict_named_module[key])
+        ):
+            setattr(dict_named_module[key][-1], "is_last_layer", True)
+            return True
+    return False
+
+
+@check_if_pytorch_greater_112()
+class BetterTransformer(object):
+    r"""
+    A conversion wrapper that takes as an input the `transformers` model to be converted
+    and returns the converted `BetterTransformer` model. The `BetterTransformer` model is based on the `BetterTransformer`
+    recently released by PyTorch from its 1.12 version:
+    https://pytorch.org/blog/a-better-transformer-for-fast-transformer-encoder-inference/
+
+    Original PR from: https://github.com/huggingface/transformers/pull/19553 adapted and wrapped in
+    this script.
+    """
+
+    def transform(
+        model: torch.nn.Module, keep_original_model: bool = False, max_memory: Optional[dict] = None, **kwargs
+    ) -> torch.nn.Module:
+        r"""
+        Conversion script from `transformers` model to its BetterTransformers version
+
+        Args:
+            model, (`torch.nn.Module`, **required**):
+                Original `transformers` model
+            keep_original_model (`bool`, *optional*):
+                whether to keep or override the original model - essentially
+                for memory efficiency reasons
+            max_memory (`dict`, *optional*):
+                Same argument as `max_memory` argument from `.from_pretrained` function
+                in `transformers`.
+        Returns:
+            The converted model if the conversion has been successful.
+        """
+
+        # Check if we have to load the model using `accelerate`
+        if hasattr(model, "hf_device_map"):
+            load_accelerate = True
+        else:
+            load_accelerate = False
+
+        hf_config = model.config
+
+        if keep_original_model:
+            model_fast = deepcopy(model)
+            model_fast = replace_to_fast(model_fast, hf_config).eval()
+        else:
+            model_fast = replace_to_fast(model, hf_config).eval()
+
+        successfully_converted_model = set_last_layer(model_fast)
+        if not successfully_converted_model:
+            raise NotImplementedError(
+                f"The Better Transformers implementation for the model {model.__class__.__name__} has not been"
+                f"implemented yet. Please open an issue requesting the addition of this model with its `BetterTransformer`"
+                f"implementation."
+            )
+
+        # Step 6: Add a class arguments, we might need to identify whether the model
+        # has been correctly converted to its `BetterTransformer` version.
+        setattr(model_fast, "is_fast", True)
```

Comment:
I would use `use_bettertransformer_fastpath` or `use_bettertransformer` or something like this

Reply:
Agreed yes. Let's be consistent here